eng/pipelines/templates/steps/compilation.check.yml

```diff
@@ -0,0 +1,34 @@
+parameters:
+  - name: Artifact
+    type: object
+    default: {}
+  - name: ArtifactName
+    type: string
+    default: "not-specified"
+  - name: ArtifactDirectory
+    type: string
+    default: "not-specified"
+  - name: RetryLimit
+    type: number
+    default: 30
+
+steps:
+  - download: current
+    artifact: ${{parameters.ArtifactName}}-signed
+  - pwsh: |
+      $pomf = Get-ChildItem "${{parameters.ArtifactDirectory}}/*.pom"
+      if(($pomf.Count -ne 1) -Or ($pomf.name -notmatch "${{parameters.Artifact.name}}")) {
+        Write-Host "Can't find package ${{parameters.Artifact.name}}"
```

Comment:
I think this error message is slightly misleading, because "can't find package" is different from "found too many packages." Also it should be `Write-Error`.